### PR TITLE
Cover gaps for accessor_iterator C++ naming requirements

### PR DESCRIPTION
**Pull request by aleksmesh**
Add checks for C++ named requirements: LegacyRandomAccessIterator for const_iterator, reverse_iterator and const_reverse_iterator with sycl::accessor, sycl::host_accessor and sycl::local_accessor

### CODE REVIEW

**Comment by aleksmesh** [comment]
CI failed due to old version of DPC++ compiler

File: tests/accessor/accessor_iterator_requirement.cpp [hunk 1]
```diff
@@ -31,47 +33,129 @@ inline void print_errors(
...
             legacy_random_access_iterator_requirement{}.is_satisfied_for(
                 dummy_acc_it);
         if (!verification_result.first) {
-          for (int i = 0; i < size_of_res_array; ++i) {
-            if (!verification_result.second[i].empty()) {
-              // Copy errors to the host side
-              res_acc[i] = verification_result.second[i];
-            }
-          }
+          //          print_errors(verification_result.second);
```
Comment (ProGTX) [inline tests/accessor/accessor_iterator_requirement.cpp:88]:
```suggestion
```

**Review by ProGTX — APPROVED** [review]

**Comment by bader** [comment]
> CI failed due to old version of DPC++ compiler

Does https://github.com/KhronosGroup/SYCL-CTS/pull/398 address this issue?

**Comment by aleksmesh** [comment]
> > CI failed due to old version of DPC++ compiler
> 
> Does #398 address this issue?

 Yes, CI passes now.

File: tests/accessor/accessor_iterator_requirement.cpp [hunk 2]
```diff
@@ -2,10 +2,12 @@
 //
 //  SYCL 2020 Conformance Test Suite
 //
-//  Provide verification that accessor's iterator is conforming to named
+//  Provide verification that accessor's iterators is conforming to named
```
Comment (keryell) [inline tests/accessor/accessor_iterator_requirement.cpp:5]:
```suggestion
//  Provide verification that accessor's iterators are conforming to named
```

**Comment by bader** [comment]
I assume that @keryell has no additional comments, so it should be good to go.

**Pull request merged**